overlord/snapstate/snapstate.go

```diff
@@ -1349,6 +1349,13 @@ func UpdateMany(ctx context.Context, st *state.State, names []string, userID int
 	return updateManyFiltered(ctx, st, names, userID, nil, flags, "")
 }
 
+// EnforceSnaps installs/updates/removes snaps reported by validErr. validationSets is the list of sets passed by the user and it's used in the final
+// stage to update validation-sets tracking in the state.
+func EnforceSnaps(ctx context.Context, st *state.State, validationSets []string, validErr *snapasserts.ValidationSetsValidationError, userID int) (tasksets []*state.TaskSet, names []string, err error) {
```

Comment:
It's a bit unusual to pass an error to another function. If you already know what kind of data this function needs (is it a list of snaps?), I'd change the type to match that.

Another option is to continue passing the error, but also make an interface for it, and have `EnforceSnaps` take the interface as parameter.

Reply:
I agree it's unusual, but it's also a very special error and it has three maps that will be needed inside the function: 	MissingSnaps map[string]map[snap.Revision][]string, InvalidSnaps map[string][]string and WrongRevisionSnapsmap[string]map[snap.Revision][]string.

While I can pass them explicitly or in a struct, I'm not sure it's better than passing ValidationSetsValidationError?

Reply:
passing an error is fine I think, but  a maybe call the argument:  validationErrorToSolve